api/clients/factory_test.go

```diff
@@ -0,0 +1,94 @@
+// @license
+// Copyright 2023 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clients
+
+import (
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/oauth2/clientcredentials"
+	"testing"
+)
+
+func TestClientCreation(t *testing.T) {
+	assert := assert.New(t)
```

Comment:
any deeper reason to go with an assert struct over the functional style of `assert.NoError(t, err)`?
I don't see any real benefit of this way tbh

Reply:
not really it just got generated like this :P
i will change it

Reply:
🤣 people write some weird code for AI to learn from...